src/m365/entra/commands/m365group/m365group-user-set.ts

```diff
@@ -83,7 +102,15 @@ class EntraM365GroupUserSetCommand extends GraphCommand {
           return `${args.options.groupId} is not a valid GUID`;
         }
 
-        if (['Owner', 'Member'].indexOf(args.options.role) === -1) {
+        if (args.options.ids && args.options.ids.split(',').some(e => !validation.isValidGuid(e))) {
+          return `${args.options.ids} contains one or more invalid GUIDs`;
+        }
+
+        if (args.options.userNames && args.options.userNames.split(',').some(e => !validation.isValidUserPrincipalName(e))) {
+          return `${args.options.userNames} contains one or more invalid usernames`;
+        }
+
+        if (['owner', 'member'].indexOf(args.options.role.toLowerCase()) === -1) {
```

Comment:
Let's change indexOf to some, also, let's initialize the allowed roles at the top of the function.